lib/libmuser.c

```diff
@@ -1295,77 +1295,87 @@ muser_ioctl(lm_ctx_t *lm_ctx, struct muser_cmd *cmd)
 }
 
 static int
-drive_loop(lm_ctx_t *lm_ctx)
+process_request(lm_ctx_t *lm_ctx)
 {
     struct muser_cmd cmd = { 0 };
     int err;
 
-    do {
-        /*
-         * FIXME for non-blocking socket we allow the user to call get_request.
-         * In SPDK that would be done in the poll function.
-         */
-        err = transports_ops[lm_ctx->trans].get_request(lm_ctx->fd, &cmd);
-        if (err < 0) {
-            lm_log(lm_ctx, LM_ERR, "failed to receive request: %m\n");
-            return err;
-        }
-        if (err == 0) {
-            if (errno == 0) {
-                lm_log(lm_ctx, LM_INF, "VFIO client closed connection\n");
-            } else {
-                lm_log(lm_ctx, LM_ERR, "end of file: %m\n");
-            }
-            break;
+    err = transports_ops[lm_ctx->trans].get_request(lm_ctx->fd, &cmd);
+    if (err < 0) {
+        lm_log(lm_ctx, LM_ERR, "failed to receive request: %m\n");
+        return err;
+    }
+    if (err == 0) {
+        if (errno == 0) {
+            lm_log(lm_ctx, LM_INF, "VFIO client closed connection\n");
+        } else {
+            lm_log(lm_ctx, LM_ERR, "end of file: %m\n");
         }
+        return -1;
```

Comment:
If the transport disconnects we'll return -1 to the caller of `lm_ctx_drive`/`lm_ctx_poll`, however that's not actual error. Maybe reset `errno` before returning to distingiush between the two cases?